donkeycar/parts/telemetry.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Telemetry class distributing real-time metrics to external server
+
+author: @miro (Meir Tseitlin) 2020
+
+Note:
+"""
+import os
+import queue
+import time
+import json
+from logging import StreamHandler
+from paho.mqtt.client import Client as MQTTClient
+
+
+class Telemetry(StreamHandler):
+    '''
+    Telemetry class collects telemetry from different parts of the system and periodically sends updated to the server.
+    Telemetry reports are timestamped and stored in memory until it is pushed to the server
+    '''
+
+    def __init__(self, cfg, default_inputs = [], default_types = []):
+
+        StreamHandler.__init__(self)
+
+        self.PUBLISH_PERIOD = cfg.TELEMETRY_PUBLISH_PERIOD
+        self._last_publish = time.time()
+        self._telem_q = queue.Queue()
+
+        self._default_inputs = default_inputs
+        self._default_types = default_types
```

Comment:
Not sure this is being used yet - see comment below.

Reply:
Not sure I understand - the whole code here is used.

Reply:
resolving this, since no comment posted